libnetwork/endpoint_cnt.go

```diff
@@ -0,0 +1,102 @@
+package libnetwork
+
+import (
+	"encoding/json"
+	"sync"
+
+	"github.com/docker/docker/libnetwork/datastore"
+)
+
+// endpointCnt was used to refcount network-endpoint relationships. It's
+// unused since v28.1, and kept around only to ensure that users can properly
+// downgrade.
+//
+// TODO(aker): remove this struct in v30.
+type endpointCnt struct {
+	n        *Network
+	Count    uint64
+	dbIndex  uint64
+	dbExists bool
+	sync.Mutex
```

Comment:
nit: It's clearer to make the mutex a separate field instead of embedding that directly into `endpointCnt`:

```suggestion
	mut sync.Mutex
```

Reply:
Yes, that's old/reverted code.

As the count's never actually updated any more and the pointer to the `Network` is effectively immutable, the mutex may not be needed - probably including for `dbIndex`. But, I don't think it's worth taking any risks at this point - and it's code that needs to be deleted again anyway.